ext/dom/xpath.c

```diff
@@ -32,179 +32,85 @@
 
 #ifdef LIBXML_XPATH_ENABLED
 
-static void dom_xpath_ext_function_php(xmlXPathParserContextPtr ctxt, int nargs, int type) /* {{{ */
+void dom_xpath_objects_free_storage(zend_object *object)
 {
-	zval retval;
-	int result, i;
-	int error = 0;
-	zend_fcall_info fci;
-	xmlXPathObjectPtr obj;
-	char *str;
-	zend_string *callable = NULL;
-	dom_xpath_object *intern;
+	dom_xpath_object *intern = php_xpath_obj_from_obj(object);
+
+	zend_object_std_dtor(&intern->dom.std);
+
+	if (intern->dom.ptr != NULL) {
+		xmlXPathFreeContext((xmlXPathContextPtr) intern->dom.ptr);
+		php_libxml_decrement_doc_ref((php_libxml_node_object *) &intern->dom);
+	}
 
+	php_dom_xpath_callbacks_dtor(&intern->xpath_callbacks);
+}
 
-	if (! zend_is_executing()) {
+HashTable *dom_xpath_get_gc(zend_object *object, zval **table, int *n)
+{
+	dom_xpath_object *intern = php_xpath_obj_from_obj(object);
+	return php_dom_xpath_callbacks_get_gc_for_whole_object(&intern->xpath_callbacks, object, table, n);
+}
+
+static void dom_xpath_proxy_factory(xmlNodePtr node, zval *child, dom_object *intern, xmlXPathParserContextPtr ctxt)
+{
+	(void) ctxt;
+
+	ZEND_ASSERT(node->type != XML_NAMESPACE_DECL);
+
+	php_dom_create_object(node, child, intern);
+}
+
+static dom_xpath_object *dom_xpath_ext_fetch_intern(xmlXPathParserContextPtr ctxt)
+{
+	if (!zend_is_executing()) {
 		xmlGenericError(xmlGenericErrorContext,
 		"xmlExtFunctionTest: Function called from outside of PHP\n");
-		error = 1;
```

Comment:
Early return to have the happy path without any indentation?

Also maybe ``UNEXPECTED()`` ?